src/reducers/MapReducer.js

```diff
@@ -47,10 +54,71 @@ const INITIAL_STATE = {
     },
     mapLoaded: false,
     viewStyle: map.SearchOptions.MAP,
-    sites: []
+    sites: [],
+    displaySites: [],
+    filterCriteriaKeys: {accessibility: [], facilities: [], price: []}
+};
+
+const updateFilterKeys = ({filterCriteriaKeys}, filterKey) => {
+    let filterCriteriaSubKey = null;
+
+    if (_.includes(_.keys(accessibility_options), filterKey)) {
+        filterCriteriaSubKey = 'accessibility';
+    } else if (_.includes(_.keys(facilities_options), filterKey)) {
+        filterCriteriaSubKey = 'facilities';
+    } else if (_.includes(_.keys(price_options), filterKey)) {
+        filterCriteriaSubKey = 'price';
+    }
+
+    const keyIsAlreadyInList = _.includes(filterCriteriaKeys[filterCriteriaSubKey], filterKey);
+    let updatedFilterKeyList = _.cloneDeep(filterCriteriaKeys);
+
+    if (keyIsAlreadyInList) {
+        updatedFilterKeyList[filterCriteriaSubKey] = _.reject(filterCriteriaKeys[filterCriteriaSubKey], existingFilterKey => existingFilterKey === filterKey);
+    } else {
+        updatedFilterKeyList[filterCriteriaSubKey] = _.concat(filterCriteriaKeys[filterCriteriaSubKey], filterKey);
+    }
+
+    return updatedFilterKeyList;
```

Comment:
If we already have the `filterKey` in our list, we should toggle it off. If we don't have it, we should add it as something to filter by.